src/chains/ethereum/ethereum/tests/api/eth/createAccessList.test.ts

```diff
@@ -0,0 +1,473 @@
+        const transaction = {
+          from,
+          to: contractAddress,
+          data
+        };
+
+        const { accessList } = await provider.send("eth_createAccessList", [
+          transaction,
+          "latest"
+        ]);
+
+        assert.strictEqual(
+          accessList.length,
+          10,
+          `Unexpected size of access list: ${accessList.length}. Expected: 1`
+        );
+        accessList.forEach((entry, i) => {
+          const addressBn = Quantity.toBigInt(to);
+          const idxBn = Quantity.toBigInt(i);
+          const sumBn = addressBn + idxBn;
+          const address = Address.from(Quantity.toBuffer(sumBn)).toString();
+          assert.strictEqual(
+            entry.address,
+            address,
+            `Access list included incorrect address. Expected: ${address}. Actual: ${entry.address}.`
+          );
+        });
+      });
+
+      it("does not create a transaction on the chain", async () => {
+        // any old transaction will do
+        const transaction = { from, to };
+        const beforeCall = await provider.send("eth_getBlockByNumber", [
+          "latest"
+        ]);
+        await provider.send("eth_createAccessList", [transaction, "latest"]);
+        const afterCall = await provider.send("eth_getBlockByNumber", [
+          "latest"
+        ]);
+        assert.strictEqual(beforeCall.number, afterCall.number);
+      });
+
+      it("returns and empty access list if EIP-2930 isn't activated", async () => {
+        // make a transaction that would generate and access list if eip was activated
+        const data = `0x${contractMethods["getStorageAt(uint256)"]}${slot}`;
+        const transaction = {
+          from,
+          to: contractAddress,
+          data
+        };
+
+        // before EIP-2930 (berlin hardfork)
+        const customProvider = await getProvider({
+          chain: { hardfork: "byzantium" }
+        });
+
+        const { accessList } = await customProvider.send(
+          "eth_createAccessList",
+          [transaction, "latest"]
+        );
+        assert.strictEqual(
+          accessList.length,
+          0,
+          `Access list has content before EIP-2930 is activated.`
+        );
+        await customProvider.disconnect();
+      });
+
+      it("returns VM exceptions", async () => {
+        const data = `0x${contractMethods["getBalance(address)"]}${encodedTo}`;
+        const transaction = {
+          from,
+          to: contractAddress,
+          data,
+          gas: "0x5208" // not enough gas
+        };
+
+        const prom = provider.send("eth_createAccessList", [
+          transaction,
+          "latest"
+        ]);
+        const callResult = {
+          execResult: {
+            runState: { programCounter: 0 },
+            exceptionError: new VmError("out of gas" as any),
+            returnValue: BUFFER_EMPTY
+          }
+        } as any;
```

Comment:
`as EVMResult` ?

Actually - looking into this, EVMResult requires `gasUsed` - should we just be setting a `gasUsed` value and not need a type assertion at all?